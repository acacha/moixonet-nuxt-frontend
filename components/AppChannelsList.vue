<template>
  <ul>
    <li v-for="channel in channels" :key="channel.id">{{ channel.name }}</li>
  </ul>
</template>

<script>
export default {
  name: 'VChannelsList',
  props: {
    channels: {
      type: Array,
      required: true
    }
  }
}
</script>
