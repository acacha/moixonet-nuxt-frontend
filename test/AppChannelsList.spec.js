import { mount } from '@vue/test-utils'
import AppChannelsList from '../components/AppChannelsList'
import channelsFixture from '../cypress/fixtures/channels'

// https://vue-test-utils.vuejs.org/
describe('AppChannelsList', () => {
  let opts

  const build = (options) => {
    options = options || opts
    const wrapper = mount(AppChannelsList, options)

    return {
      wrapper
    }
  }

  beforeEach(() => {
    opts = {
      propsData: {
        channels: channelsFixture
      }
    }
  })

  test('is a Vue instance', () => {
    const { wrapper } = build()
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders a list of channels', () => {
    const { wrapper } = build()
    expect(wrapper.element).toMatchSnapshot()
  })
})
