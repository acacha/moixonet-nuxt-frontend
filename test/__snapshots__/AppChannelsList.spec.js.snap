// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppChannelsList renders a list of channels 1`] = `
<ul>
  <li>
    
    Canal 1 
    <button
      data-test="remove-button"
    >
      
  Eliminar

    </button>
  </li>
  <li>
    
    Canal 2 
    <button
      data-test="remove-button"
    >
      
  Eliminar

    </button>
  </li>
  <li>
    
    Canal 3 
    <button
      data-test="remove-button"
    >
      
  Eliminar

    </button>
  </li>
</ul>
`;
